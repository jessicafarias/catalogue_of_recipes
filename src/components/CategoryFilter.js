import PropTypes from 'prop-types';
import '../styles/filter.css';

const CategoryFilter = props => {
  const categories = ['All', 'Beef', 'Breakfast', 'Chicken', 'Dessert', 'Goat', 'Lamb',
    'Miscellaneous', 'Pasta', 'Pork', 'Seafood', 'Side', 'Starter', 'Vegan', 'Vegetarian'];
  const { handleFilter } = props;

  const handleClick = event => {
    const { value } = event.target;
    if (value === 'All') {
      handleFilter(null);
    } else {
      handleFilter(value);
    }
  };

  return (
    <div className="container_filter sticky-top">
      <label htmlFor="w-100">
        <select className="select" name="category" id="select" onChange={handleClick}>
          <option disable="true" hidden>Search any keyword</option>
          {categories.map(
            category => <option value={category} key={category}>{category}</option>,
          )}
        </select>
      </label>
    </div>
  );
};

CategoryFilter.propTypes = {
  handleFilter: PropTypes.func.isRequired,
};

export default CategoryFilter;
