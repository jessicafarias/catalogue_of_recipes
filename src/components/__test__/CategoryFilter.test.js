import { BrowserRouter } from 'react-router-dom';
import React from 'react';
import '@testing-library/jest-dom';
import TestRenderer from 'react-test-renderer'; // ES6
import CategoryFilter from '../CategoryFilter';

describe('Display CategoryFilter.js', () => {
  it('renders correctly', () => {
    const tree = TestRenderer.create(
      <BrowserRouter>
        <CategoryFilter handleFilter={() => true} />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
