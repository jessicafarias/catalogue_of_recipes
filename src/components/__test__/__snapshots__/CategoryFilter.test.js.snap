// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display CategoryFilter.js renders correctly 1`] = `
<div
  className="container_filter sticky-top"
>
  <label
    htmlFor="w-100"
  >
    <select
      className="select"
      id="select"
      name="category"
      onChange={[Function]}
    >
      <option
        disable="true"
        hidden={true}
      >
        Search any keyword
      </option>
      <option
        value="All"
      >
        All
      </option>
      <option
        value="Beef"
      >
        Beef
      </option>
      <option
        value="Breakfast"
      >
        Breakfast
      </option>
      <option
        value="Chicken"
      >
        Chicken
      </option>
      <option
        value="Dessert"
      >
        Dessert
      </option>
      <option
        value="Goat"
      >
        Goat
      </option>
      <option
        value="Lamb"
      >
        Lamb
      </option>
      <option
        value="Miscellaneous"
      >
        Miscellaneous
      </option>
      <option
        value="Pasta"
      >
        Pasta
      </option>
      <option
        value="Pork"
      >
        Pork
      </option>
      <option
        value="Seafood"
      >
        Seafood
      </option>
      <option
        value="Side"
      >
        Side
      </option>
      <option
        value="Starter"
      >
        Starter
      </option>
      <option
        value="Vegan"
      >
        Vegan
      </option>
      <option
        value="Vegetarian"
      >
        Vegetarian
      </option>
    </select>
  </label>
</div>
`;
